process_report/util.py

```diff
@@ -27,21 +27,33 @@ def get_invoice_bucket():
     return s3_resource.Bucket(os.environ.get("S3_BUCKET_NAME", "nerc-invoicing"))
 
 
+def get_institute_mapping(institute_list: list):
+    institute_map = dict()
+    for institute_info in institute_list:
+        for domain in institute_info["domains"]:
+            institute_map[domain] = institute_info["display_name"]
+
+    return institute_map
+
+
 def get_institution_from_pi(institute_map, pi_uname):
-    institution_key = pi_uname.split("@")[-1]
-    institution_name = institute_map.get(institution_key, "")
+    institution_domain = pi_uname.split("@")[-1]
+    for i in range(institution_domain.count(".") + 1):
+        if institution_name := institute_map.get(institution_domain, ""):
+            break
+        institution_domain = institution_domain[institution_domain.find(".") + 1 :]
 
     if institution_name == "":
-        logger.warn(f"PI name {pi_uname} does not match any institution!")
+        print(f"Warning: PI name {pi_uname} does not match any institution!")
```

Comment:
why was this changed from logging to printing?

Reply:
Ah, must have been a mistake when I was moving some code around